gemd/entity/object/has_parameters.py

```diff
@@ -24,4 +27,20 @@ def parameters(self):
 
     @parameters.setter
     def parameters(self, parameters):
-        self._parameters = validate_list(parameters, Parameter)
+        def _template_check(x: Parameter) -> Parameter:
```

Comment:
This internal method `_template_check` is roughly identical to the above for conditions. Can we abstract it out and pass the `accept` condition as a lambda function maybe?

Reply:
The methods are very similar, but would require a closure for construction since (ignoring the type checking) they depend on `HasXXXXTemplates` and `validate_XXXX`.  There's also no mixin parent class here, though I can certainly build one (or just create a local util package) if DRY suggests this is the wiser route.

Reply:
I've created a centralized _template_check closure generator in HasTemplates, and then has HasSpec piggy back on it since I didn't want to implement it twice.